### PR TITLE
Borderless window support

### PR DESCRIPTION
**Pull request by IsaacMarovitz**

### CODE REVIEW

File: AKPlugin.swift [hunk 1]
```diff
@@ -44,6 +44,12 @@ class AKPlugin: NSObject, Plugin {
         NSCursor.unhide()
     }
 
+    func borderless() {
```
Comment (JoseMoreville) [inline AKPlugin.swift:47]: I think the name should be toggleBorderless or toggleTitleStyle

**Comment by Computron010** [comment]
Hi thanks for implementing this. I tried this on a local build and it works as expected however I noticed this behaviour:

I set my window size to be 1512x916 on my 14" MBP. This originally filled the entire screen, with the title-bar taking some of the vertical real estate. With this new change, using the same window size, the space that was previously occupied by the title-bar is now transparent. I tried increasing the height to around 920/930, but the effect is that the realized width is now decreased.

It does not appear that there are any combinations of the window size that allow the app/game to fill the entire screen when windowed. It seems the app/game sets the window size before the title-bar is hidden, so the window manager automatically scales back the window to fit inside the screen.

**Comment by JoseMoreville** [comment]
@Depal1 Do not merge yet, i still need to run some tests

**Comment by IsaacMarovitz** [comment]
> It does not appear that there are any combinations of the window size that allow the app/game to fill the entire screen when windowed. It seems the app/game sets the window size before the title-bar is hidden, so the window manager automatically scales back the window to fit inside the screen.

Yeah that seems to still be an issue. I'll look into that some more.

**Comment by Computron010** [comment]
The automatic resizing may be related to `constrainFrameRect`: https://developer.apple.com/documentation/appkit/nswindow/1419779-constrainframerect

This user appears to have the same issue as described and the answer is to override this function: https://stackoverflow.com/questions/6303377/nswindow-set-frame-higher-than-screen

I wanted to try this by swizzling in the Objective-C code but I'm not quite sure how to get access to `NSRect`.

**Pull request closed without merging**